test/Models/ItemSearchProviders/NumericIndexSpec.ts

```diff
@@ -0,0 +1,23 @@
+import NumericIndex from "../../../lib/Models/ItemSearchProviders/NumericIndex";
+
+describe("NumericIndex", function() {
+  describe("search", function() {
+    it("should return all matching IDs when the start and end values are the same", async function() {
+      const index = new NumericIndex("", { min: 0, max: 10 });
+      index.idValuePairs = Promise.resolve([
+        { dataRowId: 0, value: 0 },
+        { dataRowId: 1, value: 1 },
+        { dataRowId: 2, value: 2 },
+        { dataRowId: 3, value: 3 },
+        { dataRowId: 4, value: 3 },
+        { dataRowId: 5, value: 3 },
+        { dataRowId: 6, value: 3 },
+        { dataRowId: 7, value: 3 },
+        { dataRowId: 8, value: 9 },
+        { dataRowId: 9, value: 10 }
+      ]);
+      const ids = await index.search({ start: 3, end: 3 });
+      expect([...ids]).toEqual([3, 4, 5, 6, 7]);
+    });
```

Comment:
Could you add similar tests for:
- [ ] Start and end are different, and no values in the range [start, end] are present in the index
- [ ] Same as above but the range is lower than any values

and maybe a couple of others.

Reply:
I have added 3 more test cases to validate behavior for out of range search queries and in range queries.